infra/core/security/role.bicep

```diff
@@ -0,0 +1,21 @@
+metadata description = 'Creates a role assignment for a service principal.'
```

Comment:
Can we remove the one in the directory above? And are we going to remove the old deployment.bicep as part of this PR? Or if you want to make it a part of another PR can you create an issue and link it here so we dont forget?

Reply:
> Can we remove the one in the directory above?

Are you referring to the role.bicep file? We need to use it when configuring rbac permissions.


> And are we going to remove the old deployment.bicep as part of this PR? Or if you want to make it a part of another PR can you create an issue and link it here so we dont forget?

Created an issue and linked here.